Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -437,4 +437,202 @@ - (void) testUnavailableNetworkInterface {
+    [self listen]; // writable listener
+    
+    // save a doc on listenerDB
+    NSError* err = nil;
+    CBLMutableDocument* doc = [self createDocument: @"doc"];
+    [doc setValue: @"Tiger" forKey: @"species"];
+    [self.otherDB saveDocument: doc error: &err];
+    AssertNil(err);
+    
+    // save blob on db#1
+    CBLMutableDocument* doc1 = [self createDocument: @"doc-1"];
+    NSData* content = [@"i am a blob" dataUsingEncoding: NSUTF8StringEncoding];
+    CBLBlob* blob = [[CBLBlob alloc] initWithContentType: @"text/plain" data: content];
+    [doc1 setValue: blob forKey: @"blob"];
+    [self saveDocument: doc1];
+    
+    // save blob on db#2
+    [self deleteDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLDatabase* db2 = [self openDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLMutableDocument* doc2 = [self createDocument: @"doc-2"];
+    CBLBlob* blob2 = [[CBLBlob alloc] initWithContentType: @"text/plain" data: content];
+    [doc2 setValue: blob2 forKey: @"blob"];
+    [db2 saveDocument: doc2 error: &err];
+    AssertNil(err);
+    
+    // replicator#1
+    CBLReplicatorConfiguration* config1 = [self configWithTarget: _listener.localEndpoint
+                                                            type: kCBLReplicatorTypePushAndPull
+                                                      continuous: NO];
+    config1.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl1 = [[CBLReplicator alloc] initWithConfig: config1];
+    
+    // replicator#2
+    CBLReplicatorConfiguration* config2 = [[CBLReplicatorConfiguration alloc] initWithDatabase: db2
+                                                                                        target: _listener.localEndpoint];
+    config2.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl2 = [[CBLReplicator alloc] initWithConfig: config2];
+    
+    // get listener status
+    __block Listener* weakListener = _listener;
+    __block uint64_t maxConnectionCount = 0, maxActiveCount = 0;
+    id changeListener = ^(CBLReplicatorChange * change) {
+        Listener* listener = weakListener;
+        if (change.status.activity == kCBLReplicatorBusy) {
+            maxConnectionCount = MAX(listener.status.connectionCount, maxConnectionCount);
+            maxActiveCount = MAX(listener.status.activeConnectionCount, maxActiveCount);
+        } else if (change.status.activity == kCBLReplicatorStopped) {
+            if (change.replicator == repl1)
+                [exp1 fulfill];
+            else
+                [exp2 fulfill];
+        }
+    };
+    id token1 = [repl1 addChangeListener: changeListener];
+    id token2 = [repl2 addChangeListener: changeListener];
+    
+    // start & wait for replication
+    [repl1 start];
+    [repl2 start];
+    [self waitForExpectations: @[exp1, exp2] timeout: 5];
```

Comment:
If you plan to have several tests to test single lister and multiple replicators like this, can you add a method to facilitate this listener status check?